pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -306,3 +307,41 @@ def __init__(self, list_obj) -> None:
         self._order = list_obj.order
         self._class_type = list_obj.class_type
         super().__init__(list_obj)
+
+#==============================================================================
+class ListSort(ListMethod) :
+    """
+    Represents a call to the .sort() method.
+
+    Represents a call to the `.sort()` method, which sorts the elements of the
+    list in ascending order and modifies the original list in place. This means
+    that the elements of the original list are rearranged to be in sorted order.
+    Note that the .sort() method doesn't return any value.
+    
+    >>> a = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5]
+    >>> a.sort()
+    >>> print(a)
+    [1, 1, 2, 3, 3, 4, 5, 5, 5, 6, 9]
+
+    Parameters
+    ----------
+    list_obj : TypedAstNode
+        The list object which the method is called from.
+
+    reverse : TypedAstNode
+        Optional, reverse=True will sort the list descending. Default is reverse=False.
```

Comment:
```suggestion
    reverse : TypedAstNode, default=LiteralFalse
        If True the list is sorted in descending order.
```

Reply:
I suggested a change to the docstring but now that I've seen the body of the `__init__` method I don't think this is correct. The documented default is not the default value for the constructor of this class. Remember you need to document the class not Python's `sort` method.

```suggestion
    reverse : TypedAstNode, optional
        Argument mimicking sort's reverse parameter. This argument is unsupported so
        it should not be provided.
```

Reply:
updated 35a6df89c347a11eba1dbd61acd0317e55666844

Reply:
Does the second sentence not also apply to the `key` argument?

Reply:
yes i'll updated in the next commit

Reply:
88ab14fff2fe3ef6756a1cff5887173ba887536f